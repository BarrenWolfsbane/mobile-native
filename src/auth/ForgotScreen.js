import React, {
  Component
} from 'react';

import { StackActions, NavigationActions } from 'react-navigation';
import FastImage from 'react-native-fast-image';

import {
  StyleSheet,
  View,
  KeyboardAvoidingView,
  Keyboard,
  Animated,
  Platform,
} from 'react-native';
import * as Animatable from 'react-native-animatable';

import VideoBackground from '../common/components/VideoBackground';
import ForgotPassword from './ForgotPassword';
import ResetPassword from './ResetPassword';
import { CommonStyle } from '../styles/Common';
import { ComponentsStyle } from '../styles/Components';

const LOGO_HEIGHT = 100;
const LOGO_HEIGHT_SMALL = 50;

/**
 * Forgot screen
 */
export default class ForgotScreen extends Component {

  /**
   * Disable navigation bar
   */
  static navigationOptions = {
    header: null
  }

  constructor(props) {
    super(props);

    this.logoHeight = new Animated.Value(LOGO_HEIGHT);
  }

  componentWillMount () {
    this.keyboardWillShowSub = Keyboard.addListener('keyboardWillShow', this.keyboardWillShow);
    this.keyboardWillHideSub = Keyboard.addListener('keyboardWillHide', this.keyboardWillHide);
  }

  componentWillUnmount() {
    this.keyboardWillShowSub.remove();
    this.keyboardWillHideSub.remove();
  }

  keyboardWillShow = (event) => {
    Animated.timing(this.logoHeight, {
      duration: event.duration,
      toValue: LOGO_HEIGHT_SMALL,
    }).start();
  };

  keyboardWillHide = (event) => {
    Animated.timing(this.logoHeight, {
      duration: event.duration,
      toValue: LOGO_HEIGHT,
    }).start();
  };

  /**
   * Render
   */
  render() {
    const resizeMode = 'center';
    const code = this.props.navigation.state.params && this.props.navigation.state.params.code;
    console.log(this.props.navigation.state)
    return (
      <KeyboardAvoidingView style={CommonStyle.flexContainer} behavior={ Platform.OS == 'ios' ? 'padding' : null }>
        <VideoBackground />
        <View style={[CommonStyle.flexContainerCenter, CommonStyle.padding2x]}>
          <Animatable.View animation="bounceIn">
            <Animated.Image
              resizeMode="contain"
              style={[styles.logo, { height: this.logoHeight }]}
              source={require('../assets/logos/logo-white.png')}
            />
          </Animatable.View>
          <Animatable.View animation="fadeInUp">
            {code ? <ResetPassword
              onBack={this.onForgotBack}
            /> :
            <ForgotPassword
              onBack={this.onForgotBack}
            />}
          </Animatable.View>
        </View>
      </KeyboardAvoidingView>
    );
  }

  /**
   * On press back
   */
  onForgotBack = () => {
    this.props.navigation.goBack();
  }
}

const styles = StyleSheet.create({
  logo: {
    width: 200,
    height: 84,
    marginBottom: 30,
    alignSelf: 'center',
  },
});