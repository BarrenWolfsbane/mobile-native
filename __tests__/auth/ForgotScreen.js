import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import ForgotScreen from '../../src/auth/ForgotScreen';

jest.mock('../../src/auth/AuthService');
jest.mock('../../src/common/components/VideoBackground', () => 'VideoBackground');

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('ForgotScreen component', () => {

  it('should renders correctly', () => {
    const forgotScreen = renderer.create(
      <ForgotScreen />
    ).toJSON();
    expect(forgotScreen).toMatchSnapshot();
  });
});
