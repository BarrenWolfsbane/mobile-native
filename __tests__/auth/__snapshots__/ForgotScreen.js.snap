// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotScreen component should renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <VideoBackground />
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "center",
        },
        Object {
          "padding": 10,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.3,
            },
          ],
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "alignSelf": "center",
            "height": 100,
            "marginBottom": 30,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateY": 100,
            },
          ],
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
              },
              Object {
                "fontSize": 14,
              },
            ]
          }
        >
          To request a new password, enter your username
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Username"
          placeholderTextColor="white"
          returnKeyType="done"
          style={
            Array [
              Object {
                "backgroundColor": "#f8f8f8",
                "borderRadius": 4,
                "color": "#444",
                "fontSize": 16,
                "fontWeight": "600",
                "height": 50,
                "letterSpacing": 2,
                "padding": 15,
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
              Object {
                "marginTop": 10,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#FFF",
                  "borderWidth": 1,
                  "marginLeft": 10,
                  "marginRight": 0,
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "rgba(0,0,0, 0.5)",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                          "letterSpacing": 1.25,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  GO BACK
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "marginLeft": 15,
                  "marginRight": 15,
                },
                undefined,
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#FFF",
                  "borderWidth": 1,
                  "marginLeft": 10,
                  "marginRight": 0,
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={null}
            >
              <View
                pointerEvents="box-only"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#9E9E9E",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "padding": 19,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "backgroundColor": "rgba(0,0,0, 0.5)",
                    },
                    Object {
                      "borderRadius": 4,
                    },
                    Object {
                      "padding": 12,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "white",
                          "fontSize": 20,
                        },
                        undefined,
                        Object {
                          "fontSize": 17.5,
                        },
                        undefined,
                        Object {
                          "fontWeight": "600",
                          "letterSpacing": 1.25,
                        },
                        undefined,
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  CONTINUE
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
